editoast/src/views/train_schedule/mod.rs

```diff
@@ -393,29 +403,88 @@ async fn create_backend_request_payload(
     })
 }
 
-fn process_simulation_response(
-    response_payload: SimulationResponse,
+pub fn process_simulation_response(
+    SimulationResponse {
+        base_simulations,
+        eco_simulations,
+        speed_limits,
+        electrification_conditions,
+        ..
+    }: SimulationResponse,
 ) -> Vec<SimulationOutputChangeset> {
     let mut simulation_outputs = Vec::new();
+    let electrification_conditions_iter = if electrification_conditions.is_empty() {
```

Comment:
Wrong naming. It's a vec, not an iterator. You can use into_iter, though.

Reply:
fixed